src/fparser/Fortran2003.py

```diff
@@ -4200,16 +4219,18 @@ class Mask_Expr(Base): # R748
 
 class Masked_Elsewhere_Stmt(StmtBase): # R749
     """
-    <masked-elsewhere-stmt> = ELSEWHERE ( <mask-expr> ) [ <where-construct-name> ]
+    <masked-elsewhere-stmt> = ELSEWHERE
+                              ( <mask-expr> ) [ <where-construct-name> ]
     """
+    import re
     subclass_names = []
     use_names = ['Mask_Expr', 'Where_Construct_Name']
+
     @staticmethod
     def match(string):
-        if string[:9].upper()!='ELSEWHERE' and \
-           string[:10].upper() != "ELSE WHERE":
+        if not Elsewhere_Stmt._regex.match(string):
             return
-        idx = string[:10].upper().index("WHERE")
+        idx = string.upper().index("WHERE")
```

Comment:
:10 has been been removed as we may have a arbitrary number of space between ELSE and WHERE and :10 assumes a maximum of one space. Good.